dg-core/src/main/java/org/finra/datagenerator/engine/scxml/SCXMLFrontier.java

```diff
@@ -80,14 +89,109 @@ public SCXMLFrontier(final PossibleState possibleState, final SCXML model) {
+
+        if (flag.get()) {
+            return stringBuilder;
+        }
+
+        TransitionTarget nextState = state.nextState;
+
+        maxNumberOfLines = singleThreadedProcessing.getMaximumNumberOfLines();
+
+        if (nextState.getId().equalsIgnoreCase("end")) {
+            String[] numOfLines = stringBuilder.toString().split("\\n");
+
+            dataPipe = singleThreadedProcessing.processOutput(state.variables);
+
+            if (numOfLines.length < maxNumberOfLines) {
+                stringBuilder.append(dataPipe.getPipeDelimited(outTemplate)).append("\n");
+            }
+        }
+
+        //run every action in series
+        List<Map<String, String>> product = new LinkedList<>();
+        product.add(new HashMap<String, String>(state.variables));
+
+        OnEntry entry = nextState.getOnEntry();
+        List<Action> actions = entry.getActions();
+
+        for (Action action : actions) {
+            for (CustomTagExtension tagExtension : tagExtensionList) {
+                if (tagExtension.getTagActionClass().isInstance(action)) {
+                    product = tagExtension.pipelinePossibleStates(action, product);
+                }
+            }
+        }
+
+        //go through every transition and see which of the products are valid, recursive searching on them
+        List<Transition> transitions = nextState.getTransitionsList();
+
+        for (Transition transition : transitions) {
+            String condition = transition.getCond();
+            TransitionTarget target = ((List<TransitionTarget>) transition.getTargets()).get(0);
+
+            for (Map<String, String> p : product) {
+                Boolean pass;
+
+                if (condition == null) {
+                    pass = true;
+                } else {
+                    //scrub the context clean so we may use it to evaluate transition conditional
+                    Context context = this.getRootContext();
+                    context.reset();
+
+                    //set up new context
+                    for (Map.Entry<String, String> e : p.entrySet()) {
+                        context.set(e.getKey(), e.getValue());
+                    }
+
+                    //evaluate condition
+                    try {
+                        pass = (Boolean) this.getEvaluator().eval(context, condition);
+                    } catch (SCXMLExpressionException ex) {
+                        pass = false;
+                    }
+                }
+
+                //transition condition satisfied, continue search recursively
+                if (pass) {
+                    PossibleState result = new PossibleState(target, p);
+                    dfs(singleThreadedProcessing, flag, result, outTemplate, stringBuilder);
+                }
+            }
+        }
+
+        return stringBuilder;
+    }
+
+    private void dfs(Queue<Map<String, String>> queue, AtomicBoolean flag, PossibleState state) throws IOException {
```

Comment:
The processing strategy was supposed to replace the queues by hiding the queue behind a processing strategy interface. Instead of one unified frontier that is agnostic to single or multithreads, we have two separate implementations side by side.